### PR TITLE
Prohibit unsafe `as` cast in Kotlin in favor of `as?`

### PR DESCRIPTION
**Pull request by PawelLipski**

### CODE REVIEW

File: frontend/actions/src/main/kotlin/com/virtuslab/gitmachete/frontend/actions/dialogs/OverrideForkPointDialog.kt [hunk 1]
```diff
@@ -126,8 +126,10 @@ class OverrideForkPointDialog(
               isSelected: Boolean,
               cellHasFocus: Boolean
             ): Component {
-              val commit: ICommitOfManagedBranch = value as ICommitOfManagedBranch
```
Comment (PawelLipski) [inline frontend/actions/src/main/kotlin/com/virtuslab/gitmachete/frontend/actions/dialogs/OverrideForkPointDialog.kt:129]: Here's the actual fix for #1436

Comment (mkondratek) [reply frontend/actions/src/main/kotlin/com/virtuslab/gitmachete/frontend/actions/dialogs/OverrideForkPointDialog.kt:129]: how does the dialog look like when the commit is null?

Comment (PawelLipski) [reply frontend/actions/src/main/kotlin/com/virtuslab/gitmachete/frontend/actions/dialogs/OverrideForkPointDialog.kt:129]: Huh good point

image

So the entire `Inferred commit` item should be hidden then... lemme fix

Comment (PawelLipski) [reply frontend/actions/src/main/kotlin/com/virtuslab/gitmachete/frontend/actions/dialogs/OverrideForkPointDialog.kt:129]: And `commit` param can probably only be null when `val list` is empty, lemme check... 🤔

Comment (PawelLipski) [reply frontend/actions/src/main/kotlin/com/virtuslab/gitmachete/frontend/actions/dialogs/OverrideForkPointDialog.kt:129]: Ha! nope, `commit` will be `null` many times, as `getListCellRendererComponent` for some reasons gets invoked for `index = -1` and `value = null`

Comment (PawelLipski) [reply frontend/actions/src/main/kotlin/com/virtuslab/gitmachete/frontend/actions/dialogs/OverrideForkPointDialog.kt:129]: Okay, all related issues fixed

File: frontend/actions/src/main/kotlin/com/virtuslab/gitmachete/frontend/actions/dialogs/OverrideForkPointDialog.kt [hunk 2]
```diff
@@ -89,51 +89,54 @@ class OverrideForkPointDialog(
...
-          fmt(
+      val commits = branch.commitsUntilParent + listOf(branch.forkPoint, parent.pointedCommit)
+      val comboItems = commits.filterNotNull().distinct().reversed()
+      if (comboItems.isNotEmpty()) {
+        rowCompat {
+          val rb: Cell<JBRadioButton> = radioButton(
             getString(
               "action.GitMachete.BaseOverrideForkPointAction.dialog.override-fork-point.radio-button.custom"
             ),
-            branch.forkPoint?.shortHash ?: "cannot resolve commit hash"
```
Comment (PawelLipski) [inline frontend/actions/src/main/kotlin/com/virtuslab/gitmachete/frontend/actions/dialogs/OverrideForkPointDialog.kt:111]: `fmt` was redundant here (`action.GitMachete.BaseOverrideForkPointAction.dialog.override-fork-point.radio-button.custom=Manually picked commit:`, no `{...}`s)

**Review by mkondratek — APPROVED** [review]

**Pull request merged**